.github/workflows/code_quality.yml

```
@@ -20,12 +20,14 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        toxenv: [isort, black, flake8, docs]
+        toxenv: [isort, black, flake8]
+    env:
+      PIP_CONSTRAINT: pip-constraints.txt
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: '3.10'
+          python-version: '3.11'
```

OF is Python 3.12

```suggestion
          python-version: '3.12'
```
or was there a particular reason you use 3.11?

---

hmm no, I must have missed this one

---

yep, OF itself is 3.12 only anyway